src/glmfit.jl

```diff
@@ -257,31 +257,32 @@ end
 
 deviance(m::AbstractGLM) = deviance(m.rr)
 
-function nulldeviance(m::GeneralizedLinearModel{<:GlmResp{<:Any,<:Any,L}}) where L
-    hasintercept(m) ||
-        throw(ArgumentError("model does not have an intercept, " *
-                            "null deviance cannot be interpreted"))
+function nulldeviance(m::GeneralizedLinearModel)
     r      = m.rr
     wts    = weights(r.wts)
     y      = r.y
     d      = r.d
     offset = r.offset
-    dev = zero(eltype(y))
+    hasint = hasintercept(m)
+    dev    = zero(eltype(y))
     if isempty(offset) # Faster method
         if !isempty(wts)
-            mu = mean(y, weights(wts))
+            mu = hasint ?
+                mean(y, wts) :
+                linkinv(r.link, zero(eltype(y))*zero(eltype(wts))/1)
```

Comment:
```suggestion
                linkinv(r.link, zero(eltype(y)) * zero(eltype(wts)) / 1)
```

This is `0 * 0 / 1`, isn't it?

Reply:
Yes but of the appropriate type, whereas `0 * 0 / 1` will always be a `Float64`.

Reply:
Yeah, I understood the type promotion of the zeros, but it was unclear to me why we needed to promote again for integer division.

Reply:
I divide by one because `mean` divides by the number of elements. Otherwise for `Integer` inputs the result won't be float.

(I feel like these return type computation issues are really an area that need improvement in Julia.)